assets/search.js

```diff
@@ -1,20 +1,107 @@
-document.addEventListener("DOMContentLoaded", function () {
-  const searchElement = document.getElementById("search");
-  if (searchElement) {
-    searchElement.addEventListener("blur", function () {
-      searchElement.value = "";
-      searchElement.setAttribute("placeholder", "⌘K");
-    });
+import {
+  createChannel,
+  each,
+  main,
+  on,
+  resource,
+  sleep,
+  spawn,
+  suspend,
+} from "https://esm.sh/effection@4.0.0-alpha.5";
+
+await main(function* () {
+  const input = document.getElementById("search");
+  if (!input) {
+    console.log(`Search could not be setup because input was not found.`);
+    return;
+  }
+
+  const label = input.closest("label");
+  if (!label) {
+    console.log(
+      `Search could not be setup because label element was not found.`,
+    );
+    return;
+  }
+
+  const button = input.nextElementSibling;
+  if (!button) {
+    console.log(
+      `Search could not be setup because button element was not found.`,
+    );
+    return;
+  }
 
-    document.addEventListener("keydown", function (event) {
-      if (event.metaKey && event.key === "k") {
-        event.preventDefault();
-        searchElement.focus();
-        searchElement.removeAttribute("placeholder");
+  let lastBlur;
+  yield* forEach(
+    yield* join([
+      on(input, "focus"),
+      on(button, "focus"),
+      on(input, "blur"),
+      on(button, "blur"),
+    ]),
+    function* (event) {
+      if (event.type === "blur") {
+        lastBlur = yield* spawn(function* () {
+          yield* sleep(15);
+          input.value = "";
+          input.setAttribute("placeholder", "⌘K");
+          input.classList.remove("focused");
+        });
+      } else {
+        if (lastBlur) {
+          yield* lastBlur.halt();
+        }
+        input.removeAttribute("placeholder");
+        input.classList.add("focused");
       }
-      if (event.key === "Escape") {
-        searchElement.blur();
+    },
+  );
+
+  yield* forEach(on(document, "keydown"), function* (event) {
+    if (event.metaKey && event.key === "k") {
+      event.preventDefault();
+      input.focus();
+    }
+    if (event.key === "Escape") {
+      input.blur();
+    }
+  });
+
+  yield* suspend();
```

Comment:
rather than spawning a `forEach()` + `suspend()`,  why not just use a simple for loop? It will never finish.

Reply:
for with `each`? like this?
```js
for (let event of on(document, "keydown")) {
  
}
```

This wouldn't work.

Reply:
Curious why not?

Reply:
🤔 what is the difference between `for(let event of yield* each(on(document, "keydown")) { }` vs `for(let event of on(document, "keydown") { }`

Reply:
The latter one won't work 😆 

But I was suggesting that you use the former.

Reply:
Oh, I see what you mean. I did that now. I'm using for/each and removed suspense.